tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -187,20 +188,23 @@ def __getitem__(
     flip_dims = []
     for i, idx in enumerate(index):
 
+        # t.shape could potentially return ints, so we would want to convert in those cases
+        shape_dim = DimensionSize(t_shape[i]) if not isinstance(t_shape[i], Tensor) else t_shape[i]
```

Comment:
Should we do this conversion only at the point where it's needed, i.e. in `where`?

Reply:
I'll do that. Might be simpler that way.